INTMDB-625: Add README for CFN & CDK examples (#405)
mongodb/mongodbatlas-cloudformation-resources: cdk/CONTRIBUTING.md

> # Contributing to MongoDB Atlas CDK

Author: Getting the contributing guides split out ahead of our planning repo split for cdk

MongoDB welcomes community contributions! If you’re interested in making a contribution, please follow the steps below before you start writing any code:

1. Reach out by filing an [issue](https://github.com/mongodb/mongodbatlas-cloudformation-resources/issues) to discuss your proposed contribution, be it a bug fix or feature/other improvements.
1. Sign the [contributor's agreement](http://www.mongodb.com/contributor). This will allow us to review and accept contributions.

> After the above 2 steps are completed and we've agreed on a path forward:

Reviewer: ```suggestion
After the above steps are completed and we've agreed on a path forward:
```

1. Fork the repository on GitHub
2. Create a branch with a name that briefly describes your submission
3. Add comments around your new code that explain what's happening

> 4. Commit and push your changes to your branch then submit a pull request against the current release branch, not master.  The naming scheme of the branch is `release-staging-v#.#.#`. Note: There will only be one release branch at a time.

Reviewer 2: ```suggestion
4. Commit and push your changes to your branch then submit a pull request against the current release branch, not master. The naming scheme of the branch is `release-staging-v#.#.#`. **Note**: There will only be one release branch at a time.
```

5. A repo maintainer will review the your pull request, and may either request additional changes or merge the pull request.

## Autoclose stale issues and PRs
- After 30 days of no activity (no comments or commits are on an issue or PR) we automatically tag it as “stale” and add a message: "This issue has gone 30 days without any activity and meets the project’s definition of ‘stale’. This will be auto-closed if there is no new activity over the next 30 days. If the issue is still relevant and active, you can simply comment with a “bump” to keep it open, or add the “[Status] Not Stale” label. Thanks for keeping our repository healthy!"
- After 30 more days of no activity we automatically close the issue / PR.

## CDK L1 Dev and Publishing
1. Install all the dependencies - `yarn, node, cdk-import`
2. Let's take an example For creating cdk for `private-endpoint` resource.
   Make sure cfn-resources/private-endpoint has the latest schema.
3. Execute `./cdk.sh private-endpoint`

> 4. This would generate the scaffolding using projen and

Reviewer 2: ```suggestion
4. This would generate the scaffolding using `projen` and
```

>    would also emit autogenerated index.ts in the src directory.

Reviewer 2: ```suggestion
   would also emit autogenerated `index.ts` in the src directory.
```

5. Update the README.md file as per the CDK library requirements.

## Release and Publishing
### Manual Release

> 1. projenrc is set to do manual release.

Reviewer 2: ```suggestion
1. `projenrc` is set to do manual release.
```

2. The release process is based on *Semantic Versioning* and *Conventional Commits* (https://www.conventionalcommits.org/).

> 3. Make sure to follow the above guidelines in order to bump appropriate part of the version(*MAJOR, MINOR, PATCH*).

Reviewer 2: ```suggestion
3. Make sure to follow the above guidelines in order to bump appropriate part of the version (*MAJOR, MINOR, PATCH*).
```

4. Commit the changes with the message based on the conventional commits guidelines.

### Publish

> 1. Make sure to create an NPM account and generate NPM token.

Reviewer 2: ```suggestion
1. Make sure to create an NPM account and generate an NPM token.
```

2. `export NPM_TOKEN=<>` to generated token.

> 3. You may also need to `export NPM_ACCESS_LEVEL=public` in case it is not set in projenrc

Reviewer 2: ```suggestion
3. You might also need to `export NPM_ACCESS_LEVEL=public` in case it is not set in `projenrc`.
```

4. Execute `./publish.sh private-endpoint` to publish the resource.

### Tagging

> 1. The version is bumped based on the previous release version which is the tag in the git repository.

Reviewer 2: ```suggestion
1. The version is bumped based on the previous release version, which is the tag in the git repository.
```

> 2. Release task would look for previous tag in the git repository and bump the version based on the last commit message.

Reviewer 2: add article + simpler verb form
```suggestion
2. Release task looks for the previous tag in the git repository and bump the version based on the last commit message.
```
